piptools/scripts/compile.py

```diff
@@ -115,7 +118,65 @@ def _determine_linesep(
 @options.config
 @options.no_config
 @options.constraint
+@typing.no_type_check
 def cli(
+    *args,
+    verbose: int,
+    quiet: int,
+    **kwargs,
+):
+    """
+    Compiles requirements.txt from requirements.in, pyproject.toml, setup.cfg,
+    or setup.py specs.
+    """
+    log.verbosity = verbose - quiet
+    current_python = sys.executable
+    log.debug(f"{current_python=}")
+    env_python = shutil.which("python")
+    log.debug(f"{env_python=}")
+    if current_python != env_python:
+        # pip-tools probably installed globally (e.g. via pipx)
+        # install pip-tools in a venv made by env_python and run it there
+        click.echo("Installing pip-tools in temporary venv...", err=True)
+        with tempfile.TemporaryDirectory(prefix="pip-tools-env-") as venv_path:
+            log.debug(f"Installing venv at {venv_path}")
+            run(  # nosec
+                [
+                    env_python,
+                    "-m",
+                    "venv",
+                    venv_path,
+                ],
```

Comment:
I do wonder: Could we just run `env_python -m piptools compile *sys.argv[1:]` instead of creating a nested virtualenv?